include/stl2/detail/iterator/ostreambuf_iterator.hpp

```diff
@@ -23,69 +23,68 @@
 #include <stl2/detail/iterator/default_sentinel.hpp>
 
 STL2_OPEN_NAMESPACE {
-	namespace __ostreambuf_iterator {
-		template <MoveConstructible charT, class traits>
-		class cursor {
-			using ostream_type = std::basic_ostream<charT, traits>;
-			using streambuf_type = std::basic_streambuf<charT, traits>;
-			detail::raw_ptr<streambuf_type> sbuf_ = nullptr;
-
+	// Not to spec:
+	// * MoveConstructible requirement is implicit
+	//   See https://github.com/ericniebler/stl2/issues/246)
+	// * Extension: satisfies EqualityComparable and Sentinel<default_sentinel>
+	template <MoveConstructible charT, class traits = std::char_traits<charT>>
+	class ostreambuf_iterator {
 		public:
-			class mixin : protected detail::ebo_box<cursor> {
-				using box_t = detail::ebo_box<cursor>;
-			public:
-				using difference_type = std::ptrdiff_t;
-				using char_type = charT;
-				using traits_type = traits;
-				using ostream_type = cursor::ostream_type;
-				using streambuf_type = cursor::streambuf_type;
-
-				constexpr mixin() noexcept = default;
-				using box_t::box_t;
+			typedef ptrdiff_t difference_type;
+			typedef charT char_type;
+			typedef traits traits_type;
+			typedef std::basic_streambuf<charT, traits> streambuf_type;
+			typedef std::basic_ostream<charT, traits> ostream_type;
 
-				STL2_CONSTEXPR_EXT bool failed() const noexcept {
-					return box_t::get().sbuf_ == nullptr;
-				}
-			};
-
-			constexpr cursor() noexcept = default;
-			STL2_CONSTEXPR_EXT cursor(streambuf_type* s) noexcept
-			: sbuf_{s}
-			{}
-			cursor(ostream_type& s) noexcept
-			: cursor{s.rdbuf()}
-			{}
-			// Extension
-			constexpr cursor(default_sentinel) noexcept
-			: cursor{}
-			{}
-
-			void write(charT c) {
+			ostreambuf_iterator() = default;
```

Comment:
I prefer `constexpr ostreambuf_iterator() noexcept = default;` so the compiler will shout loudly if we screw up this type's subobjects in the future.